micropython.js

```diff
@@ -147,10 +151,21 @@ class MicroPythonBoard {
 
   async run(code, data_consumer) {
     data_consumer = data_consumer || function() {}
-    await this.enter_raw_repl()
-    const output = await this.exec_raw(code || '#', data_consumer)
-    await this.exit_raw_repl()
-    return Promise.resolve(output)
+    return new Promise(async (resolve, reject) => {
+      if (this.reject_run) {
+        this.reject_run('re run')
+        this.reject_run = null
+      }
+      this.reject_run = reject
+      try {
+        await this.enter_raw_repl()
+        const output = await this.exec_raw(code || '#', data_consumer)
+        await this.exit_raw_repl()
+        return resolve(output)
+      } catch (e) {
+        reject(e)
```

Comment:
Question: if there is an error here, and the promise is rejected, then the next `run` call will find `this._reject_run` to be not `null` and will reject it, but nothing will happen, as it has already been rejected. Is this expected behavior, or would it be better to set `this.reject_run` to `null` in the `catch`?